modules/programs/hexchat.nix

```
@@ -0,0 +1,323 @@
+      (listChar "C" channel.commands)
+    ]));
+in {
+  meta.maintainers = with maintainers; [ superherointj thiagokokada ];
+
+  options.programs.hexchat = with types; {
+    enable = mkEnableOption "HexChat - Graphical IRC client";
+
+    channels = mkOption {
+      default = null;
+      type = nullOr (attrsOf modChannelOption);
+      description = "Configures '~/.config/hexchat/servlist.conf'";
+      example = literalExample ''
+        {
+          freenode = {
+            autojoin = [
+              "#home-manager"
+              "#linux"
+              "#nixos"
+            ];
+            charset = "UTF-8 (Unicode)";
+            commands = [
+              "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+            ];
+            loginMethod = sasl;
+            nickname = "my_nickname";
+            nickname2 = "my_secondchoice";
+            options = {
+              acceptInvalidSSLCertificates = false;
+              autoconnect = true;
+              bypassProxy = true;
+              connectToSelectedServerOnly = true;
+              useGlobalUserInformation = false;
+              forceSSL = false;
+            };
+            password = "my_password";
+            realName = "my_realname";
+            servers = [
+              "chat.freenode.net"
+              "irc.freenode.net"
+            ];
+            userName = "my_username";
+          };
+        }'';
+    };
+
+    settings = mkOption {
+      default = null;
+      description = ''
+        Configuration for "~/.config/hexchat/hexchat.conf", see
+        <link xlink:href="https://hexchat.readthedocs.io/en/latest/settings.html#list-of-settings"/>
+        for supported values.
+      '';
+      example = literalExample ''
+        {
+          irc_nick1 = "mynick";
+          irc_username = "bob";
+          irc_realname = "Bart Simpson";
+          text_font = "Monospace 14";
+        };
+      '';
+      type = nullOr (attrsOf str);
+    };
+
+    overwriteConfigFiles = mkOption {
+      type = nullOr bool;
+      description = ''
+        Enables overwritting HexChat configuration files (hexchat.conf, servlist.conf).
+        Any existing HexChat configuration will be lost.
+        Certify to back-up any previous configuration before enabling this.
+
+        Enabling this setting is recommended, because everytime HexChat application is
+        closed it overwrites Nix/Home-Manager provided configuration files, causing:
+        1. Nix/HM provided configuration to be out of sync with actual active HexChat configuration.
+        2. Blocking Nix/HM updates until configuration files are manually removed.
+      '';
+      default = false;
+    };
+
+    theme = mkOption {
+      default = null;
+      description = ''
+        Theme package for HexChat.
+        Expects a derivation containing decompressed theme files.
+        '.hct' file format requires unzip decompression, as seen in example.
+      '';
+      example = ''
```

```suggestion
      example = literalExample ''
```

I think it's safe to use `literalExample` with an expression (the mkderivation function call) like that.